File: java/kora/base-project/src/main/java/com/github/mroose/backend/service/SampleService.java
package com.github.mroose.backend.service;

import com.github.mroose.backend.config.client.BaseWebClientConfig;
import lombok.RequiredArgsConstructor;
import reactor.core.publisher.Flux;
import com.github.mroose.backend.dto.SampleDto;
import reactor.core.publisher.Mono;
import ru.tinkoff.kora.common.Component;

import java.util.List;

@Component
@RequiredArgsConstructor
public class SampleService {

    private final BaseWebClientConfig baseWebClient;

    public List<SampleDto> getTodos() {
        return baseWebClient.getTodos();
    }
}

File: kotlin/kora/base-project/src/main/kotlin/com/github/mroose/backend/service/SampleService.kt
package com.github.mroose.backend.service

import com.github.mroose.backend.dto.SampleDto
import org.springframework.stereotype.Service
import org.springframework.web.reactive.function.client.WebClient
import reactor.core.publisher.Flux

@Service
class SampleService(val baseWebClient: WebClient) {

    fun getTodos(): Flux<SampleDto> {
        return baseWebClient
            .get()
            .uri("http://jsonplaceholder.typicode.com/todos")
            .retrieve()
            .bodyToFlux(SampleDto::class.java)
    }
}